libs/features/mas/web-components/src/aem.js

```diff
@@ -1,63 +1,211 @@
+     * @returns {Promise<Array>} - A promise that resolves to an array of search results
+     */
+    async searchFragment({ path, query, variant }) {
+        const filter = {};
+        if (path) {
+            filter.path = path;
+        }
+        if (query) {
+            filter.fullText = {
+                text: encodeURIComponent(query),
+                queryMode: 'EXACT_WORDS',
+            };
+        }
+
+        const searchParams = new URLSearchParams({
+            query: JSON.stringify({ filter }),
+        }).toString();
+        return fetch(`${this.cfSearchUrl}?${searchParams}`, {
+            headers: this.headers,
+        })
+            .then((res) => res.json())
+            .then((json) => json.items)
+            .then((items) => {
+                if (variant) {
+                    return items.filter((item) => {
+                        const [itemVariant] = item.fields.find(
+                            (field) => field.name === 'variant',
+                        )?.values;
+                        return itemVariant === variant;
+                    });
+                }
+                return items;
+            });
+    }
+
+    /**
+     * Get fragment by path
+     * @param {string} path fragment path
+     * @returns {Promise<Object>} the raw fragment item
+     */
+    async getFragmentByPath(path) {
+        const headers = this.#author ? this.headers : {};
+        return fetch(`${this.cfFragmentsUrl}?path=${path}`, { headers })
+            .then((res) => res.json())
+            .then(({ items: [item] }) => item);
+    }
+
+    async getFragment(res) {
+        const eTag = res.headers.get('Etag');
+        const fragment = await res.json();
+        fragment.etag = eTag;
+        return fragment;
+    }
+
+    /**
+     * Get fragment by ID
+     * @param {string} id fragment id
+     * @returns {Promise<Object>} the raw fragment item
+     */
+    async getFragmentById(id) {
+        return await fetch(`${this.cfFragmentsUrl}/${id}`, {
+            headers: this.headers,
+        }).then(this.getFragment);
+    }
+
+    /**
+     * Save given fragment
+     * @param {Object} fragment
+     * @returns {Promise<Object>} the updated fragment
+     */
+    async saveFragment(fragment) {
+        const { title, fields } = fragment;
+        return await fetch(`${this.cfFragmentsUrl}/${fragment.id}`, {
+            method: 'PUT',
+            headers: {
+                'Content-Type': 'application/json',
+                'If-Match': fragment.etag,
+                ...this.headers,
+            },
+            body: JSON.stringify({ title, fields }),
+        }).then(this.getFragment);
+    }
+
+    /**
+     * Copy a content fragment using the AEM classic API
+     * @param {Object} fragment
+     * @returns {Promise<Object>} the copied fragment
+     */
+    async copyFragmentClassic(fragment) {
+        const csrfToken = await this.getCsrfToken();
```

Comment:
Should we consider caching the CSRF token to improve performance? Wondering if this changes frequently?

Reply:
Or I guess if we won't copy fragments too many times it might be okay...

Reply:
csrf token changes often and as said before, we should not use `copyFragmentClassic` but `copyFragment` once the API is fixed.